src/application/hook.ts

```diff
@@ -1,10 +1,11 @@
 import { useRecoilState } from 'recoil';
-import { dataState, loadState } from './state';
+import { itemsState } from './state';
 import { Item } from './state';
+import { useState } from 'react';
 
 const useItemFetch = () => {
-  const [data, setData] = useRecoilState<Item[]>(dataState);
-  const [showMore, setShowMore] = useRecoilState<number>(loadState);
+  const [data, setData] = useRecoilState<Item[]>(itemsState);
+  const [showMore, setShowMore] = useState(10);
```

Comment:
You cannot use useState in the hook.ts. If you use useState, it will create a new state variable every time you call `useItemFetch` 

Instead, this needs to be an atom with recoil.js